### PR TITLE
use rust relecent func, add virtualization ext 

### PR DESCRIPTION
**Pull request by pengzechen**
use rust relecent func, add virtualization ext

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,3 +13,4 @@ categories = ["embedded", "no-std", "hardware-support", "os"]
 
 [dependencies]
 tock-registers = "0.8"
+spinlock = { path = "../../crates/spinlock" }
```
Comment (hky1999) [inline Cargo.toml]: emmmmmmm

File: src/device_ref.rs
Comment (hky1999) [inline src/device_ref.rs:1]: Why do we need this file here?

File: src/gic_v2.rs [hunk 2]
```diff
@@ -111,70 +125,234 @@ pub struct GicDistributor {
...
 
 unsafe impl Send for GicDistributor {}
 unsafe impl Sync for GicDistributor {}
 
 unsafe impl Send for GicCpuInterface {}
 unsafe impl Sync for GicCpuInterface {}
 
+use crate::device_ref::*;
+
+static mut GICD: DeviceRef<GicDistributor> = unsafe { DeviceRef::new() };
```
Comment (hky1999) [inline src/gic_v2.rs:148]: Instantiate it in axhal of arceos, just as we discussed in the meeting yesterday.

File: src/gic_v2.rs [hunk 3]
```diff
@@ -111,70 +125,234 @@ pub struct GicDistributor {
...
+    /// Get gicd iid reg
+    pub fn get_iidr() -> u32 {
+        unsafe { GICD.IIDR.get() }
+    }
+    
     /// The number of implemented CPU interfaces.
-    pub fn cpu_num(&self) -> usize {
-        ((self.regs().TYPER.get() as usize >> 5) & 0b111) + 1
+    pub fn cpu_num() -> usize {
+        unsafe{ ((GICD.TYPER.get() as usize >> 5) & 0b111) + 1 }
```
Comment (hky1999) [inline src/gic_v2.rs:180]: Remove all GICD.xxx functions

**Review by hky1999 — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (pengzechen) [reply Cargo.toml]: emmm, sorry.

Comment (hky1999) [reply Cargo.toml:16]: most importantly, arm_gicv2 should not hold spinlock or dep on spinlock

File: src/device_ref.rs
Comment (pengzechen) [reply src/device_ref.rs:1]: Using gic::init_base, to init dev base addr.

**Review by hky1999 — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (hky1999) [reply Cargo.toml]: arm_gicv2 should not hold spinlock or dep on spinlock

**Comment by pengzechen** [comment]
arm_gicv2 should not hold spinlock or dep on spinlock
Why ?

**Comment by pengzechen** [comment]
The data：
image
Red Hat Inc did
image
Linux did
The lock： 
Old implementation lost too much performance, which is unacceptable.

**Pull request closed without merging**